changelog.d/1614.bugfix

```diff
@@ -0,0 +1 @@
+PublicitySyncer: Replace Object with Map to avoid risk of prototype pollution by malicious Matrix Room IDs.
```

Comment:
Needs to describe the functional change in behaviour for the end-user/admin, rather than the code change we made